### ISSUE TITLE
Upload function to upload whole folder to cloud storage

### ISSUE THREAD

**Issue by abskrj** [ZeStream/zestream-server#7]
Write a function in `utils` folder, which takes a folder path and cloud storage name (was, gcp, azure) and uploads the whole folder to the cloud storage

**Comment by scoobhidu** [comment]
Hi abhishek I would like to pickup this issue, 
Which provider would you like to use?, or should i keep it configurable acc to .env?

**Comment by scoobhidu** [comment]
also should I make it for AWS S3 as well or only azure blob and gcp bucket?

**Comment by abskrj** [comment]
it should be configurable, and any new cloud provider could be added in future

**Comment by scoobhidu** [comment]
@abhishekraj272 are these going to be temp files? gcp has an option for those as well

**Comment by abskrj** [comment]
yes, temp files, we need support for was, gcp and azure, all three

**Comment by abskrj** [comment]
@loyaltytrooper  working on this?

**Comment by scoobhidu** [comment]
Sorry abhishek i am not well from last 3 days you can remove me from this issue

**Comment by Shofiya2003** [comment]
can I work on the issue? 
we need a function which takes the path and the cloud storage name(you can add any new one in the future) and upload the folder to that storage. Right?

**Comment by abskrj** [comment]
Yes @Shofiya2003 
Assigning it it to you

**Comment by abskrj** [comment]
@Shofiya2003 working on this?

**Comment by Shofiya2003** [comment]
Yes

**Comment by Shofiya2003** [comment]
hey!
the function to upload the folder will take the folder path and the cloud platform as the parameter, and will write independent functions for each of the cloud storage. 
I am planning to use filepath.Walk  to get the files in the folder.
Apologies for the time, trying to figure things out, new to Golang

**Comment by Shofiya2003** [comment]
@abhishekraj272

**Comment by abskrj** [comment]
yes, right
you can use `filepath.Walk`

**Comment by abskrj** [comment]
let me know if you stuck somewhere

**Comment by Shofiya2003** [comment]
yo!
wrote the functions to upload on s3 bucket and gcp bucket. 
Having troubles configuring the AWS SDK to test my function. working on it

**Comment by Shofiya2003** [comment]
I used environment variables for aws access key and the secret key but it seems it cannot find the environment variables in the environment. I also used os.Getenv() function to print the keys but it does not work. Do I need some package to access the environment variables in golang?
I used this to configure the SDK, created sessions successfully but stuck on credentials.
https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html

**Comment by abskrj** [comment]
have you put these values in the `.env` file 
```
AWS_ACCESS_KEY_ID=YOUR_AKID
AWS_SECRET_ACCESS_KEY=YOUR_SECRET_KEY
AWS_SESSION_TOKEN=TOKEN
```

and are you running this function in main `configs.LoadEnv()` before running your function?

**Comment by Shofiya2003** [comment]
yes i have those values in `.env` file 
not running the function
will get back to you after making the changes

**Comment by Shofiya2003** [comment]
done with aws. thanks a ton 
I had to use configs.LoadEnv().

**Comment by abskrj** [comment]
thats great

**Comment by Shofiya2003** [comment]
hey!
created a function for the three cloud storage. I did run a test for all of them, it seems to be working fine. 
will make the changes after the review
Thank you for the help

**Comment by Shofiya2003** [comment]
@abhishekraj272

**Comment by abskrj** [comment]
that's great, also thanks for contribution :)

**Comment by abskrj** [comment]
hey @Shofiya2003,
could you take this issue after this is merged #47?

**Comment by Shofiya2003** [comment]
sure
